.github/workflows/ci.yml

```diff
@@ -32,73 +32,191 @@ jobs:
+      DOCKER_IMAGE_TAG: ${{ github.sha }}
+      DD_API_KEY: ${{ secrets.DD_CI_API_KEY }}
+      DD_APP_KEY: ${{ secrets.DD_CI_APP_KEY }}
     steps:
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: "11"
+          distribution: "adopt"
+
       - name: Checkout system tests
         uses: actions/checkout@v3
 
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@dc7b9719a96d48369863986a06765841d7ea23f6 # 2.0.0
+        with:
+          install: true
+          config-inline: |
+            [worker.oci]
+              max-parallelism = 1
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2 
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build test images
+        run: ./lib-injection/run.sh build-and-push-test-app-image build-and-push-init-image
+
       - name: Build
         run: ./lib-injection/build.sh
 
       - name: Run
-        run: ./lib-injection/run.sh
+        run: ./lib-injection/run-lib-injection.sh
+        
+      - name: Compress logs
+        if: ${{ always() }}
+        run: tar -czvf artifact.tar.gz $(ls | grep logs)
 
-  injection-tests-node:
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        if: ${{ always() }}
+        with:
+          name: library-injection_java_${{ matrix.weblog-variant }}_${{ matrix.lib-injection-connection }}_${{ matrix.lib-injection-use-admission-controller }}
+          path: artifact.tar.gz
+
+  injection-tests-python:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
     strategy:
       matrix:
-        # node does not support UDS
-        lib-injection-connection: ['network']
+        lib-injection-connection: [ 'network','uds']
         lib-injection-use-admission-controller: ['', 'use-admission-controller']
+        weblog-variant: [ 'dd-lib-python-init-test-django','dd-lib-python-init-test-django-gunicorn','dd-lib-python-init-test-django-uvicorn']
       fail-fast: false
     env:
-      TEST_LIBRARY: nodejs
+      TEST_LIBRARY: python
+      WEBLOG_VARIANT: ${{ matrix.weblog-variant }}
       LIBRARY_INJECTION_CONNECTION: ${{ matrix.lib-injection-connection }}
-      LIBRARY_INJECTION_ADMISSION_CONTROLLER: ${{ matrix.lib-injection-admission-controller }}
-      LIBRARY_INJECTION_INIT_IMAGE: ghcr.io/datadog/dd-trace-js/dd-lib-js-init:e1043d06c7b25f5494a80d3c95e43029ac5cefff
-      LIBRARY_INJECTION_TEST_APP_IMAGE: ghcr.io/datadog/dd-trace-js/sample-app:45cb74896b0b528d23b1458c9cacfa9b296658ba
+      LIBRARY_INJECTION_ADMISSION_CONTROLLER: ${{ matrix.lib-injection-use-admission-controller }}
+      DOCKER_REGISTRY_IMAGES_PATH: ghcr.io/datadog/system-tests
+      DOCKER_IMAGE_TAG: ${{ github.sha }}
+      BUILDX_PLATFORMS: linux/amd64
+      DD_API_KEY: ${{ secrets.DD_CI_API_KEY }}
```

Comment:
As per our discussions relating to secrets. 
I think it might be better to scope this to only be exposed for the command that needs it.

I presume its `./lib-injection/run-lib-injection.sh` - this way the surface area where the secret could leak is narrower